applications/ShapeOptimizationApplication/tests/shape_optimization_test_factory.py

```diff
@@ -7,6 +7,12 @@
 # Other imports
 import os
 
+# Check if external Apps are available
+has_eigen_app =  kratos_utilities.CheckIfApplicationsAvailable("EigenSolversApplication")
+has_csm_app = kratos_utilities.CheckIfApplicationsAvailable("StructuralMechanicsApplication")
+has_mesh_moving_app = kratos_utilities.CheckIfApplicationsAvailable("MeshMovingApplication")
+
+# Define controlled execution scope
 class controlledExecutionScope:
```

Comment:
can you please use the class from KratosUnittest here?

Reply:
which one?